src/esp/physics/bullet/BulletRigidObject.cpp

```diff
@@ -138,6 +144,7 @@ bool BulletRigidObject::initialization_LibSpecific(
 
   //! Add to world
   bWorld_->addRigidBody(bObjectRigidBody_.get());
+  collisionObjToObjIds_->emplace(bObjectRigidBody_.get(), objectId_);
```

Comment:
Nit: as an alternative to doing this collision-object-to-habitat-id mapping yourself, there is btCollisionObject::setUserIndex/getUserIndex. This is commonly how Bullet user code will retrieve a "game object ID" or similar from a bullet collision object returned by a query like a raycast.

Reply:
Cool, thanks for pointing that out. Will consider as a future update. 👍